lib/using_restricted_array.rb

```diff
@@ -6,56 +6,125 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
```

Comment:
Since you don't allocate more data as n increases in size, this is O(1) for space complexity.